app/controllers/proxy_subnets_controller.rb

```diff
@@ -0,0 +1,32 @@
+class ProxySubnetsController < ApplicationController
+  before_filter :find_proxy, :only => [:index, :details]
+
+  def index
+    @subnets = @smart_proxy.statuses[:dhcp].subnets
+    render :partial => 'smart_proxies/plugins/dhcp_subnets', :locals => { :subnets => @subnets }
+  rescue Foreman::Exception => e
+    process_ajax_error e
+  end
+
+  def details
```

Comment:
I would rename it show for consistency?

also, now that you have a separate controller (great!) i think it would be easy to implement delete too (see the dhcp_browser plugin for trivial reference)